app/src/main/java/com/Found404/paypro/ui/pages/AddingMerchantsPage.kt

```diff
@@ -42,4 +86,86 @@ fun AddingMerchants(
             Icon(imageVector = Icons.Default.Add, contentDescription = "Add")
         }
     }
-}
\ No newline at end of file
+}
+
+@Composable
+fun MerchantItem(merchant: MerchantResponse, onDeleteTerminal: (String) -> Unit) {
+    var showPopup by remember { mutableStateOf(false) }
+    var selectedTerminalId by remember { mutableStateOf("") }
+
+    Box(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(vertical = 8.dp, horizontal = 16.dp)
+            .background(color = Color.LightGray, shape = RoundedCornerShape(16.dp))
+            .padding(horizontal = 16.dp, vertical = 8.dp)
+    ) {
+        Column(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalAlignment = Alignment.Start,
+            verticalArrangement = Arrangement.SpaceBetween
+        ) {
+            Text(
+                text = "Your Merchant",
+                style = TextStyle(fontSize = 14.sp),
+                modifier = Modifier.padding(bottom = 8.dp)
+            )
+            Text(text = merchant.merchantName, color = Color.Black, style = TextStyle(fontSize = 30.sp))
+            Text(text = "${merchant.address.streetName}, ${merchant.address.city}")
+            Text(text = "Street No: ${merchant.address.streetNumber}, Postal Code: ${merchant.address.postalCode}")
+
+            merchant.terminals.forEach { terminal ->
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                    horizontalArrangement = Arrangement.SpaceBetween
+                ) {
+                    Text(text = "Terminal: ${terminal.terminalKey}")
+                    Icon(
+                        imageVector = Icons.Default.Delete,
+                        contentDescription = "Delete Terminal",
+                        modifier = Modifier
+                            .clickable {
+                                selectedTerminalId = terminal.terminalKey
+                                showPopup = true
+                            }
+                    )
+                }
+            }
+        }
+    }
+
+    if (showPopup) {
+        DeleteTerminalPopup(
+            terminalId = selectedTerminalId,
+            onConfirm = {
+                onDeleteTerminal(selectedTerminalId)
+                showPopup = false
+            },
+            onCancel = { showPopup = false }
+        )
+    }
+}
+@Composable
+fun DeleteTerminalPopup(terminalId: String, onConfirm: () -> Unit, onCancel: () -> Unit) {
```

Comment:
Please place this composable in separate file in order to keep this page cleaner. You put it in `components` and call it `DeleteTerminalModal` or `DeleteTerminalPopup`